deeprankcore/dataset.py

```diff
@@ -474,32 +476,40 @@ def __init__( # pylint: disable=too-many-arguments
         """
         super().__init__(hdf5_path, subset, target, task, classes, tqdm, root, target_filter, check_integrity)
 
+        self.default_vars = {
+            "features": "all",
+            "target": None,
+            "target_transform": False,
+            "task": None,
+            "classes": None
+            }
```

Comment:
I was also looking for something like this, but `__annotations__` gives all the parameters, and `__defaults__` only the ones defaulted. This causes a shift in the assignments since `hdf5_path` does not have a default. I also don't like to hardcode the parameters, but I couldn't find another reliable way. I'll give another look at it and in case I'll ask your review again

Reply:
good catch!
maybe not as elegant, but how about:

```suggestion
        first_def = len(self.__init__.__annotations__) - len(self.__init__.__defaults__)
        self.default_vars = {
            var: default for var, default in zip(list(self.__init__.__annotations__)[first_def:], self.__init__.__defaults__)
            }
```

Reply:
or, more elegant:

```suggestion
        ds = self.__init__.__defaults__
        vs = list(self.__init__.__annotations__)[-len(ds):]

        default_vars = {var: default for var, default in zip(vs, ds)}
```

Reply:
Nicely packed! The risk that I keep seeing here is that if for reasons we change the second parameter (for example), and we decide that it doesn't have to default to anything anymore, then this code won't work (bc we need to exclude the first two parameters instead of only the first) and it will be difficult to remember to change `default_vars`. Hard coding is ugly, but at least keeps very explicit which parameters defaulted you're considering and is less error-prone for future developers. 
What would always work instead, would be something that automatically takes only the parameters defaulted, with their default value, independently from how many parameters are mandatory or defaulted to something. I was looking for smt like this, so like `__init__.__defaults__` but with parameters names as well, but the closest thing I found was the one you've already suggested

Reply:
the way i wrote it it'll still work if we change the number of values that are defaulted. his is because the defaulted ones always precede the non-defaulted ones and `vs` is sliced using `[-len(ds):]` and non-defaulted arguments cannot follow defaulted ones.

[Alternative option](https://stackoverflow.com/a/12627202/5170442):

```python
import inspect

signature = inspect.signature(self.__init__)
default_vars = {
    k: v.default
    for k, v in signature.parameters.items()
    if v.default is not inspect.Parameter.empty
}
```

Reply:
You're right, nice! I didn't notice it.
I really like the alternative option, it is indeed more explicit. Both are fine at this point, but I'll go with the second one :)